### PR TITLE
aepp update boilerplate

### PR DESCRIPTION
**Pull request by thepiwo**
code is not nicely formatted or in any way improved, eslint issues will still have to be fixed

### CODE REVIEW

File: governance-aepp/src/utils/contract.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import pollContractSource from '../assets/contracts/Poll.aes';
+import pollIrisContractSource from '../assets/contracts/Poll_Iris.aes';
+import {toRefs} from "vue";
+import {sdk, wallet} from "@/utils/wallet";
+
+const contract = {
+  registry: null
+};
+
+// TODO adjust ACI usage
```
Comment (thepiwo) [inline governance-aepp/src/utils/contract.js:12]: will do that in a later PR for contracts/server at the same time

File: governance-aepp/src/utils/contract.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }));
+};
+
+contract.polls = async () => {
+  await contract.init();
+  const polls = await contract.registry.methods.polls();
+  return Array.from(polls.decodedResult.entries());
+};
+
+contract.verifyPollContract = async (pollAddress) => {
```
Comment (thepiwo) [inline governance-aepp/src/utils/contract.js:46]: Currently this is still the old way of doing this, haven't tried out the new possibilities as exposed with source hash now. Maybe will do this in a secondary step.

File: governance-aepp/src/components/PollListing.vue [hunk 3]
```diff
@@ -46,6 +44,10 @@
         height: 0
       };
     },
+    setup() {
+      const {networkId} = toRefs(wallet)
+      return {networkId}
+    },
```
Comment (kenodressel) [inline governance-aepp/src/components/PollListing.vue:50]: afaik this is only needed when you use the networkId in the template

File: governance-aepp/src/utils/contract.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import {sdk, wallet} from "@/utils/wallet";
+
+const contract = {
+  registry: null
+};
+
+// TODO adjust ACI usage
+
+contract.init = async () => {
+  const {networkId} = toRefs(wallet)
```
Comment (kenodressel) [inline governance-aepp/src/utils/contract.js:15]: why not access `wallet.networkId` below?

File: governance-aepp/src/utils/contract.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      delegator: delegator,
+      delegatee: delegatee,
+      delegatorAmount: await sdk.getBalance(delegator),
+      includesIndirectDelegations: delegateeDelegations.length !== 0,
+    };
+  }));
+};
+
+contract.polls = async () => {
+  await contract.init();
```
Comment (kenodressel) [inline governance-aepp/src/utils/contract.js:41]: shouldn't we check if the contract is already initialized?

File: governance-aepp/src/utils/contract.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  const polls = await contract.registry.methods.polls();
+  return Array.from(polls.decodedResult.entries());
+};
+
+contract.verifyPollContract = async (pollAddress) => {
+  const {networkId} = toRefs(wallet)
```
Comment (kenodressel) [inline governance-aepp/src/utils/contract.js:47]: lets use wallet.networkId

File: governance-aepp/src/utils/wallet.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  } catch (error) {
+    console.error('initWallet error:', error)
+    return false
+  }
+
+  return true
+}
+
+export const scanForWallets = async () => {
+  const {walletStatus, activeWallet} = toRefs(wallet)
```
Comment (kenodressel) [inline governance-aepp/src/utils/wallet.js:59]: should not be needed

File: governance-aepp/src/utils/wallet.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+}]
+
+export let sdk = null
+
+export const wallet = reactive({
+  activeWallet: null, address: null, balance: null, walletStatus: null, isStatic: false, networkId: null,
+})
+
+export const initWallet = async (eventBus) => {
+  const {walletStatus, isStatic, networkId} = toRefs(wallet)
```
Comment (kenodressel) [inline governance-aepp/src/utils/wallet.js:21]: should not be needed

File: governance-aepp/src/utils/wallet.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+      const scannerConnection = new BrowserWindowMessageConnection()
+      const stopScan = walletDetector(scannerConnection, handleWallets)
+    })
+  } else {
+    walletStatus.value = 'fallback_static'
+  }
+}
+
+export const fetchWalletInfo = async () => {
+  const {address, balance, walletStatus, isStatic, networkId} = toRefs(wallet)
```
Comment (kenodressel) [inline governance-aepp/src/utils/wallet.js:107]: if we access the wallet object directly it should not be needed

**Review by kenodressel — COMMENTED** [review]
looks good in general. I'd say we should revisit the `useRefs` including pattern of the wallet as it should only be required when we are looking to import a specific property that will be used in the template part.

A simpler setup for these cases would be:
```
    setup: () => wallet, 
```
or
```
    setup: () => { wallet }, 
```

File: governance-aepp/.nvmrc [hunk 10]
```diff
@@ -1 +0,0 @@
-14.20.0
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply governance-aepp/.nvmrc:1]: it was just a temporary measure, currently it should be supported by all modern versions, thus removed

File: governance-aepp/README.md [hunk 11]
```diff
@@ -1,17 +1,35 @@
-# Governance Aepp Vue.JS Frontend
+# Boilerplate Aepp Vue.JS Frontend
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply governance-aepp/README.md:1]: yeah will have to fix that back up

File: governance-aepp/README.md [hunk 12]
```diff
@@ -1,17 +1,35 @@
-# Governance Aepp Vue.JS Frontend
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply governance-aepp/README.md:1]: yes, copy paste error

File: governance-aepp/src/components/PollListing.vue (hunk 3 above)
Comment (thepiwo) [reply governance-aepp/src/components/PollListing.vue:50]: I want to use it elsewhere with `this.` seems to work as is

File: governance-aepp/src/utils/contract.js (hunk 4 above)
Comment (thepiwo) [reply governance-aepp/src/utils/contract.js:15]: if that works I can try

File: governance-aepp/src/utils/contract.js (hunk 5 above)
Comment (thepiwo) [reply governance-aepp/src/utils/contract.js:41]: Bug I introduced last minute when preparing for network change, this shouldn't be needed any more

File: governance-aepp/src/utils/wallet.js (hunk 9 above)
Comment (thepiwo) [reply governance-aepp/src/utils/wallet.js:107]: is it still reactive then? this is the pattern I followed from the boilerplate

Comment (thepiwo) [reply governance-aepp/src/utils/wallet.js:107]: https://github.com/aeternity/aepp-boilerplate-vue/blob/master/src/utils/aeternity/wallet.js

**Review by kenodressel — APPROVED** [review]

**Pull request merged**